src/extension.ts

```diff
@@ -64,7 +51,7 @@ export async function activate(context: vscode.ExtensionContext): Promise<void>
           vscode.ConfigurationTarget.Global,
         );
       }
-      await updateLanguageServer(manifest.version, lsPath);
+      await clientHandler.startClient();
       return clientHandler.startClient();
```

Comment:
When starting the client via the 'Enable Language Server' command here, we aren't catching any errors and are not running `vscode.commands.executeCommand('setContext', 'terraform.showTreeViews', true);` afterwards.

This is resulting in two different error messages:
_here_
<img width="400" alt="CleanShot 2022-01-17 at 14 35 50@2x" src="https://user-images.githubusercontent.com/45985/149779016-192c60e8-6063-4248-935c-6f95f966e309.png">

_line 135_
<img width="508" alt="CleanShot 2022-01-17 at 14 36 11@2x" src="https://user-images.githubusercontent.com/45985/149779017-504b4a77-de9e-4040-9a70-622fe9cdff9c.png">

As you just added the error dialog in line 135, do you think we should do this here, too? Or should I create an issue for that?

Reply:
I extracted two helper methods to wrap calling start/stop with error handling to improve the enable/disable commands. I don't like them being there in the first place, but that is another set of work entirely.

Speaking of new issues, we need to add error handling with discovering the binary to avoid errors like you showed. This generic error handler won't precisely tell the user what to do and we can do better than that.